app/Main.hs

```diff
@@ -64,45 +64,46 @@ formatOne ::
   Maybe FilePath ->
   IO ExitCode
 formatOne CabalDefaultExtensionsOpts {..} mode config mpath =
-  withPrettyOrmoluExceptions (cfgColorMode config) $
-    case FP.normalise <$> mpath of
-      Nothing -> do
-        extraDynOptions <-
-          if optUseCabalDefaultExtensions
-            then case optStdinInputFile of
-              Just stdinInputFile ->
-                getCabalExtensionDynOptions stdinInputFile
-              Nothing -> throwIO OrmoluMissingStdinInputFile
-            else pure []
-        r <- ormoluStdin (configPlus extraDynOptions)
-        case mode of
-          Stdout -> do
-            TIO.putStr r
-            return ExitSuccess
-          _ -> do
-            hPutStrLn
-              stderr
-              "This feature is not supported when input comes from stdin."
-            -- 101 is different from all the other exit codes we already use.
-            return (ExitFailure 101)
-      Just inputFile -> do
-        extraDynOptions <-
-          if optUseCabalDefaultExtensions
-            then getCabalExtensionDynOptions inputFile
-            else pure []
-        originalInput <- readFileUtf8 inputFile
-        formattedInput <- ormoluFile (configPlus extraDynOptions) inputFile
+  let (inputFile, extraDynOptions) = case FP.normalise <$> mpath of
+        Nothing ->
+          ( stdinPath,
+            if optUseCabalDefaultExtensions
+              then case optStdinInputFile of
+                Just cabalStdinInputFile ->
+                  getCabalExtensionDynOptions cabalStdinInputFile
+                Nothing -> throwIO OrmoluMissingStdinInputFile
+              else pure []
+          )
+        Just realInputFile ->
+          ( realInputFile,
+            if optUseCabalDefaultExtensions
+              then getCabalExtensionDynOptions realInputFile
+              else pure []
```

Comment:
The logic `if optUseCabalDefaultExtensions then getCabalExtensionDynOptions ... else pure []` could be shared.